File: DungeonDebt/Assets/Scripts/Data/EncounterDefinition.cs
using System.Collections.Generic;
using System.Collections.ObjectModel;

public class EncounterDefinition
{
    public EncounterDefinition(
        int act,
        int slot,
        EncounterType type,
        string displayName,
        string scoutText,
        string dangerCategory,
        IList<EnemyDefinition> enemies,
        int baseGoldReward,
        EncounterEffectId encounterEffectId,
        RivalGuild rivalGuild)
    {
        Act = act;
        Slot = slot;
        Round = GameRules.GetAbsoluteRound(act, slot);
        Type = type;
        DisplayName = displayName;
        ScoutText = scoutText;
        DangerCategory = dangerCategory;
        Enemies = new ReadOnlyCollection<EnemyDefinition>(new List<EnemyDefinition>(enemies));
        BaseGoldReward = baseGoldReward;
        EncounterEffectId = encounterEffectId;
        RivalGuild = rivalGuild;
    }

    public int Act { get; }
    public int Slot { get; }
    public int Round { get; }
    public EncounterType Type { get; }
    public string DisplayName { get; }
    public string ScoutText { get; }
    public string DangerCategory { get; }
    public IReadOnlyList<EnemyDefinition> Enemies { get; }
    public int BaseGoldReward { get; }
    public EncounterEffectId EncounterEffectId { get; }
    public RivalGuild RivalGuild { get; }
}

File: web/src/data/EncounterDefinition.js
// Ported from DungeonDebt/Assets/Scripts/Data/EncounterDefinition.cs
import { GameRulesFns } from "../core/GameRules.js";

export class EncounterDefinition {
  constructor(act, slot, type, displayName, scoutText, dangerCategory, enemies, baseGoldReward, encounterEffectId, rivalGuild, variantId = "base", enemyBoardPositions = null) {
    this.act = act;
    this.slot = slot;
    this.round = GameRulesFns.getAbsoluteRound(act, slot);
    this.variantId = variantId;
    this.type = type;
    this.displayName = displayName;
    this.scoutText = scoutText;
    this.dangerCategory = dangerCategory;
    this.enemies = Object.freeze([...enemies]);
    this.baseGoldReward = baseGoldReward;
    this.encounterEffectId = encounterEffectId;
    this.rivalGuild = rivalGuild;
    // Optional array of {q,r} authored board positions per enemy slot (index-matched
    // to enemies[]). Null entries fall back to the default enemy deployment mapping.
    this.enemyBoardPositions = enemyBoardPositions ? Object.freeze([...enemyBoardPositions]) : null;
    Object.freeze(this);
  }
}
